components/CheckBox/CheckBox.tsx

```diff
@@ -65,15 +71,45 @@ function CheckBox({ filters, handleFilters }) {
     } else {
       newCheck.splice(currentId, 1);
     }
+
     setChecked(newCheck);
     localStorage.setItem("checked", JSON.stringify(newCheck));
     handleFilters(newCheck);
   };
 
-  function isChecked(id) {
-    return !(Checked.indexOf(id) === -1);
+
+  const allChecked = (index: number) => {
+    return !(event[index].some((elem) => { return !Checked.includes(elem.id) }))
+
+  }
+  const noneChecked = (index: number) => {
+    return !event[index].some((elem) => { return Checked.includes(elem.id) })
   }
 
+  const handleToggleAll = (values, index: number, index1: number) => {
```

Comment:
For a cleaner way to do this function, you can use:

```js
  const handleToggleAll = (values, index: number, index1: number) => {
    let newChecked = [];

    // If all checkboxed are not checked, check them all.
    // Else, newChecked array will continue being empty and will uncheck them all
    if (!isAllChecked(index1 + sync[index])) {
      newChecked = values.map((event) => event.id);
    }
    
    setChecked(newChecked);
    handleFilters(newChecked);
    localStorage.setItem("checked", JSON.stringify(newChecked));
  };
```
That way is more concise and readable.